### DISCUSSION TITLE
Adjoint solver with INC_SOLVER and periodic boundaries

### DISCUSSION THREAD

**Discussion by hareshramn** [su2code/SU2#1424] [Q&A] [answered]
Hello,
            I'm trying to run a shape optimization simulation for a 2D external flow over a hydrofoil using INC_RANS and Discrete adjoint solver. I have periodic boundary conditions in one direction. 
The flow solver works fine and I get a converged solution. 
I get a segmentation fault error when I run the adjoint solver using SU2_CFD_AD.
I have attached a screenshot of the error. 
![Untitled](https://user-images.githubusercontent.com/22233134/140189356-68c882bc-f698-4fa9-b587-7aea13b4f7b5.jpg)

I also find that the error goes away when I use Outlet BCs instead of the periodic BCs.

Any help will be appreciated.
Thanks!

**Comment by TobiKattmann** [comment, marked as answer]
Hi, can you please attach the case with mesh if possible? 

You can also take a look [here](https://su2code.github.io/tutorials/Inc_Streamwise_Periodic/) which is a INC_RANS with periodic case where a working Discrete adjoint is available. At the bottom there is a link to a 2D CHT case where a `README.md` explains a bit more. I.e. to my knowledge there is no problem per se with INC_RANS + periodic +DiscAdj

In case you compiled the code yourself. Please attach `--buildtype=debug` to the `meson.py ...` command. That way you will get much better information where the segfault occurs in the code

**Comment by hareshramn** [comment]
Thank you for the reply. I have attached my mesh and cfg file.
[planar_2D.zip](https://github.com/su2code/SU2/files/7490971/planar_2D.zip)
 
I was able to run the shape optimization script without the segmentation fault when I updated to version 7.2.1 and SA turbulence model. Have not really found the source of my issue though.

I will compile the code with the debug flag on and check.

Thanks for the info!

**Reply by TobiKattmann** [reply]
👍 If you cannot reproduce the segfault on the latest version and you are happy with the solution, I guess we could call it day :D 
I am not aware of any fixes to the periodic bc lately ... but lets wait for your results
